inference-engine/src/transformations/src/transformations/serialize.cpp

```diff
@@ -0,0 +1,374 @@
+    pugi::xml_node m_data;
+
+    template <typename T>
+    std::string create_atribute_list(
+        ngraph::ValueAccessor<std::vector<T>>& adapter) {
+        return joinVec(adapter.get(), std::string(","));
+    }
+
+public:
+    XmlVisitor(pugi::xml_node& data) : m_data(data) {}
+
+    void on_adapter(const std::string& name,
+                    ngraph::ValueAccessor<void>& adapter) override {
+#if 0  // TODO: remove when Constant will support VisitorAPI
+        m_data.append_attribute(name.c_str());
+#endif
+    }
+    void on_adapter(const std::string& name,
+                    ngraph::ValueAccessor<bool>& adapter) override {
+        m_data.append_attribute(name.c_str()).set_value(adapter.get());
+    }
+    void on_adapter(const std::string& name,
+                    ngraph::ValueAccessor<std::string>& adapter) override {
+        m_data.append_attribute(name.c_str()).set_value(adapter.get().c_str());
+    }
+    void on_adapter(const std::string& name,
+                    ngraph::ValueAccessor<int64_t>& adapter) override {
+        m_data.append_attribute(name.c_str()).set_value(adapter.get());
+    }
+    void on_adapter(const std::string& name,
+                    ngraph::ValueAccessor<double>& adapter) override {
+        m_data.append_attribute(name.c_str()).set_value(adapter.get());
+    }
+    void on_adapter(
+        const std::string& name,
+        ngraph::ValueAccessor<std::vector<int64_t>>& adapter) override {
+        m_data.append_attribute(name.c_str())
+            .set_value(create_atribute_list(adapter).c_str());
+    }
+    void on_adapter(
+        const std::string& name,
+        ngraph::ValueAccessor<std::vector<uint64_t>>& adapter) override {
+        m_data.append_attribute(name.c_str())
+            .set_value(create_atribute_list(adapter).c_str());
+    }
+    void on_adapter(
+        const std::string& name,
+        ngraph::ValueAccessor<std::vector<float>>& adapter) override {
+        m_data.append_attribute(name.c_str())
+            .set_value(create_atribute_list(adapter).c_str());
+    }
+    void on_adapter(
+        const std::string& name,
+        ngraph::ValueAccessor<std::vector<std::string>>& adapter) override {
+        m_data.append_attribute(name.c_str())
+            .set_value(create_atribute_list(adapter).c_str());
+    }
+};
+
+const std::map<ngraph::Node*, int> create_layer_ids(const ngraph::Function& f) {
+    std::map<ngraph::Node*, int> layer_ids;
+    int id = 0;
+    for (auto node : f.get_ordered_ops()) {
+        layer_ids[node.get()] = id++;
+    }
+    return layer_ids;
+}
+
+const std::vector<Edge> create_edge_mapping(
+    const std::map<ngraph::Node*, int>& layer_ids, const ngraph::Function& f) {
+    std::vector<Edge> edges;
+    for (auto node : f.get_ordered_ops()) {
+        if (dynamic_cast<ngraph::op::Parameter*>(node.get()) != nullptr) {
+            continue;
+        }
+
+        for (auto i : node->inputs()) {
```

Comment:
```suggestion
        for (const auto& i : node->inputs()) {
```

Reply:
Done in this [commit](https://github.com/openvinotoolkit/openvino/pull/2579/commits/09fbad0067999b153c2c78be728aafef935d25aa)